fclient/crosssigning.go

```diff
@@ -46,6 +47,49 @@ type CrossSigningKey struct {
 
 func (s *CrossSigningKey) isCrossSigningBody() {} // implements CrossSigningBody
 
+func (s *CrossSigningKey) Equal(other *CrossSigningKey) bool {
+	if s == nil || other == nil {
+		return false
+	}
+	if s.UserID != other.UserID {
+		return false
+	}
+	if len(s.Usage) != len(other.Usage) {
+		return false
+	}
+	for i := range s.Usage {
+		if s.Usage[i] != other.Usage[i] {
```

Comment:
Is this slice guaranteed to be ordered? If not you might get some false negatives.

Reply:
Fair point, but does `usage` currently even contain more than one value?

Reply:
Per spec no, but you want to be forwards compatible, otherwise you might need to prevent such cross signing keys from being uploaded.

Reply:
Fixed in https://github.com/matrix-org/gomatrixserverlib/pull/444/commits/30cbdb4f140c61ae8a5c233bbe869aeb61610d27